extensions/ql-vscode/src/query-history/store/domain-mapper.ts

```diff
@@ -0,0 +1,89 @@
+import { LocalQueryInfo, InitialQueryInfo } from "../../query-results";
+import { QueryEvaluationInfo } from "../../run-queries-shared";
+import { QueryHistoryInfo } from "../query-history-info";
+import {
+  LocalQueryDataItem,
+  InitialQueryInfoData,
+  QueryEvaluationInfoData,
+} from "./local-query-data-item";
+import { QueryHistoryDataItem } from "./query-history-data";
+import { VariantAnalysisDataItem } from "./variant-analysis-data-item";
+
+// Maps Query History Domain Models to Data Models
+
+export function mapQueryHistoryToDataModels(
+  queries: QueryHistoryInfo[],
+): QueryHistoryDataItem[] {
+  return queries.map((q) => {
+    if (q.t === "variant-analysis") {
+      const query: VariantAnalysisDataItem = q;
+      return query;
+    } else if (q.t === "local") {
+      return mapLocalQueryInfoToDataModel(q);
+    } else {
+      throw new Error("Query type not supported");
```

Comment:
```suggestion
      assertNever(q);
```
I think we would be better off using `assertNever` here because then the type system will enforce that we can't reach this branch. That's a bit safer than having the type system allow it but then failing at runtime. Since for this mapper the values are coming from the application rather than from serialized data, we can safely trust that the type information will be accurate.

Reply:
👍